test/test-utils.dart

```diff
@@ -68,6 +68,18 @@ class SuperTokensTestUtils {
     return request;
   }
 
+  static http.Request getLoginRequestUtf8Encoded() {
+    var loginAPIURL = "$baseUrl/login";
+    var request = http.Request('POST', Uri.parse(loginAPIURL));
+    request.headers['Content-Type'] = "application/json; charset=utf-8";
+    var body = {"userId": "supertokens-flutter-tests", "payload": {
+      "name": "\xc3\xb6\xc3\xa4\xc3\xbc\x2d\xc3\xa1\xc3\xa0\xc3\xa2" // UTF-8 encoded öäü-áàâ
```

Comment:
i don't think the login enpoint [here](https://github.com/supertokens/supertokens-flutter/blob/10eb3dae5875819bce43771bfa0d7f69abb6447b/testHelpers/server/index.js#L212) actually uses the `payload` prop in the reuqest to add to the access token payload of the session. So that probably needs to be modified as well

Reply:
Ah yes, that's a mixup with `/login-2.18`, where the request payload gets passed to the token. Would it be fine to run the test against `/login-2.18`? Or should I rather adapt the `/login` node-function?

In the second case, I guess I would need to pass the payload like so: `Session.createNewSession(req, res, userId, accessTokenPayload: tokenFromRequest);`?

Reply:
correct. You should adapt the login API and pass in the `payloadFromRequest` to the createNewSession function in case the request body has this prop